lib/pkgdb.sh

```diff
@@ -0,0 +1,297 @@
+    set -- $(
+        for pkg in "$@" ; do
+            package_name=$(distro_name_package "$pkg")
+            # When we could not find a different package name for the distribution
+            # we're running on we try the package name given as last attempt
+            if [ -z "$package_name" ]; then
+                package_name="$pkg"
+            fi
+            echo "$package_name"
+        done
+        )
+
+    case "$SPREAD_SYSTEM" in
+        ubuntu-*|debian-*)
+            apt-get remove -y --purge -y "$@"
+            ;;
+        fedora-*)
+            dnf -y remove "$@"
+            dnf clean all
+            ;;
+        opensuse-*)
+            zypper remove -y "$@"
+            ;;
+        arch-*)
+            pacman -Rnsc --noconfirm "$@"
+            ;;
+        amazon-*)
+            yum -y remove "$@"
+            yum clean all
+            ;;
+        *)
+            echo "ERROR: Unsupported distribution $SPREAD_SYSTEM"
+            exit 1
+            ;;
+    esac
+}
+
+distro_update_package_db() {
+    case "$SPREAD_SYSTEM" in
+        ubuntu-*|debian-*)
+            apt-get update
+            ;;
+        fedora-*)
+            dnf clean all
+            dnf makecache
+            ;;
+        opensuse-*)
+            zypper --gpg-auto-import-keys refresh
+            ;;
+        arch-*)
+            pacman -Syq
+            ;;
+        amazon-*)
+            yum clean all
+            yum --nogpgcheck makecache
+            ;;
+        *)
+            echo "ERROR: Unsupported distribution $SPREAD_SYSTEM"
+            exit 1
+            ;;
+    esac
+}
+
+distro_upgrade_packages() {
+    case "$SPREAD_SYSTEM" in
+        ubuntu-*|debian-*)
+            echo "so far not needed"
+            ;;
+        fedora-*)
+            echo "so far not needed"
+            ;;
+        opensuse-*)
+            echo "so far not needed"
+            ;;
+        arch-*)
+            pacman --needed --noconfirm -Syu
+            ;;
+        amazon-*)
+            yum -y update
+            ;;
+        *)
+            echo "ERROR: Unsupported distribution $SPREAD_SYSTEM"
+            exit 1
+            ;;
+    esac
+}
+
+distro_clean_package_cache() {
+    case "$SPREAD_SYSTEM" in
+        ubuntu-*|debian-*)
+            apt-get clean
+            ;;
+        fedora-*)
+            dnf clean all
+            ;;
+        opensuse-*)
+            zypper -q clean --all
+            ;;
+        arch-*)
+            pacman -Sccq --noconfirm
+            ;;
+        amazon-*)
+            yum clean all
+            ;;
+        *)
+            echo "ERROR: Unsupported distribution $SPREAD_SYSTEM"
+            exit 1
+            ;;
+    esac
+}
+
+distro_auto_remove_packages() {
+    case "$SPREAD_SYSTEM" in
+        ubuntu-*|debian-*)
+            apt-get -y autoremove
+            ;;
+        fedora-*)
+            dnf -y autoremove
+            ;;
+        opensuse-*)
+            ;;
+        arch-*)
+            ;;
```

Comment:
We could probably do `pacman -Rnsc --noconfirm $(pacman -Qdtq)`. This will remove all packages that were installed as dependencies but now have become orphans.